src/atoms/Button/Button.native.js

```diff
@@ -1,35 +1,216 @@
+      case 'secondary':
+        return variantColor ? theme.colors[variantColor][500] : theme.colors.primary[500];
+      case 'tertiary':
+        return variantColor ? theme.colors[variantColor][500] : theme.colors.primary[500];
+    }
+  },
+  padding({ size, theme, text }) {
+    switch (size) {
+      case 'xsmall':
+        if (!text) {
+          return `${theme.spacings.xxs}`;
+        }
+        return `${theme.spacings.xxs} ${theme.spacings.s}`;
+      case 'small':
+        if (!text) {
+          return `${themeHelper.px(0.75)}`;
+        }
+        return `5px ${theme.spacings.l}`; //TODO: vertical padding is 5px in Figma, unit is 8
+      case 'medium':
+      default:
+        if (!text) {
+          return `${theme.spacings.s}`;
+        }
+        return `${theme.spacings.s} ${theme.spacings.xxl}`;
+      case 'large':
+        if (!text) {
+          return `${themeHelper.px(1.25)}`;
+        }
+        return `${themeHelper.px(1.25)} ${theme.spacings.xxl} `;
+    }
+  },
+  iconSize(size) {
+    switch (size) {
+      case 'xsmall':
+        return 'xsmall';
+      case 'small':
+        return 'small';
+      case 'medium':
+      case 'large':
+      default:
+        return 'medium';
+    }
+  },
+};
+
+const StyledButton = styled(TouchableHighlight)((props) => {
+  return `
+    flex-direction: row;
+    align-items: center;
+    align-self: ${props.align};
+    background-color: ${styles.backgroundColor(props)};
+    padding: ${styles.padding(props)};
+    border-radius: ${props.theme.spacings.xxs};
+    ${styles.border(props) && `border: ${styles.border(props)}`};
+  `;
+});
+
+const StyledText = styled(Text)((props) => {
+  return `
+    color: ${styles.fontColor(props)};
+    letter-spacing: ${props.theme.fonts.letterSpacing.l};
+  `;
+});
+
+const Button = ({
+  onClick,
+  children,
+  variant,
+  size,
+  disabled,
+  variantColor,
+  icon,
+  iconPosition,
+  align,
+  tesId,
+}) => {
+  const theme = useContext(ThemeContext);
+  return (
+    <StyledButton
+      onPress={onClick}
+      variantColor={variantColor}
+      disabled={disabled}
+      size={size}
+      variant={variant}
+      activeOpacity={1}
+      text={children}
+      underlayColor={styles.underlayColor({ variantColor, variant, theme })}
+      align={align}
+      tesId={tesId}
+    >
+      <>
+        {icon && iconPosition === 'left' && (
+          <Icon
+            name={icon}
+            size={styles.iconSize(size)}
+            fill={styles.fontColor({ variantColor, variant, disabled, theme })}
+          />
+        )}
+        {icon && iconPosition === 'left' && children && (
+          <Space margin={[0, 0.5]}>
+            <View />
+          </Space>
+        )}
+        <StyledText variantColor={variantColor} variant={variant} disabled={disabled}>
+          {children}
+        </StyledText>
+        {icon && iconPosition === 'right' && children && (
+          <Space padding={[0, 0.5]}>
+            <View />
```

Comment:
Instead of having an empty `View` can we have the `Space` component as a `View` by default?